src/Shipment/Request/PostReturnShipmentsRequest.php

```diff
@@ -0,0 +1,137 @@
+     */
+    public function __construct(ShipmentCollection $collection, array $parameters = [])
+    {
+        $this->collection = $collection;
+        $this->parameters = $parameters;
+    }
+
+    /**
+     * @return string
+     */
+    public function getBody(): string
+    {
+        return json_encode([
+            'data' => [
+                'return_shipments' => $this->encodeReturnShipments(),
+            ],
+        ]);
+    }
+
+    /**
+     * @return array
+     */
+    public function getHeaders(): array
+    {
+        return [
+            'Content-Type' => 'application/vnd.return_shipment+json;charset=utf-8',
+        ];
+    }
+
+    /**
+     * @return string
+     */
+    public function getHttpMethod(): string
+    {
+        return 'POST';
+    }
+
+    /**
+     * @return string
+     */
+    public function getPath(): string
+    {
+        return $this->path;
+    }
+
+    /**
+     * @return array
+     */
+    protected function getQueryParameters(): array
+    {
+        return array_filter($this->parameters);
+    }
+
+    /**
+     * @param  \MyParcelNL\Pdk\Shipment\Model\Shipment $shipment
+     *
+     * @return null|array
+     * @throws \MyParcelNL\Pdk\Base\Model\InvalidCastException
+     */
+    private function encodeReturnOptions(Shipment $shipment): array
+    {
+        $shipmentOptions = $shipment->deliveryOptions->shipmentOptions;
+        $options         = array_map(static function ($item) {
+            return is_bool($item) ? (int) $item : $item;
+        }, $shipmentOptions->toSnakeCaseArray());
+
+        return array_filter(
+            [
+                'package_type' => $this->getPackageTypeId($shipment),
+                'insurance'    => $shipmentOptions->insurance
+                    ? [
+                        'amount'   => $shipmentOptions->insurance * 100,
+                        'currency' => 'EUR',
+                    ] : null,
+            ] + $options
+        );
+    }
+
+    /**
+     * @return array
+     */
+    private function encodeReturnShipments(): array
+    {
+        return $this->collection->map(function ($shipment) {
+            return [
+                'parent'               => $shipment->id,
+                'reference_identifier' => $shipment->referenceIdentifier,
+                'carrier'              => $shipment->carrier->id,
+                'email'                => $shipment->recipient->email,
+                'name'                 => $shipment->recipient->person,
+                'options'              => $this->encodeReturnOptions($shipment),
+            ];
+        })->all();
+    }
+
+    /**
+     * @param  \MyParcelNL\Pdk\Shipment\Model\Shipment $shipment
+     *
+     * @return int
+     */
+    private function getPackageTypeId(Shipment $shipment): int
+    {
+        return DeliveryOptions::PACKAGE_TYPES_NAMES_IDS_MAP[$shipment->deliveryOptions->packageType];
```

Comment:
dit kan een error gooien

Reply:
Is dat zo, want een shipment heeft toch altijd een packageType attribuut in DeliveryOptions zitten?

Reply:
het kan zijn dat dat niet een waarde is die in `DeliveryOptions::PACKAGE_TYPES_NAMES_IDS_MAP` zit